### ISSUE TITLE
Http Sender validation fails when using new "Use Map" option with x-www-form-urlencoded Content-Type

### ISSUE THREAD

**Issue by is-simon** [nextgenhealthcare/connect#4395]
When using the new "Use Map" option of the HTTP Sender, the validation of the channel fails, without any real useful error message. I had to dig in into the source code to understand what was happening... 

The validation done [here](https://github.com/nextgenhealthcare/connect/blob/development/client/src/com/mirth/connect/connectors/http/HttpSender.java#L641) should verify if the "Use Map" option has been used instead of the parameters map.

**Comment by is-simon** [comment]
If someone has the same problem...

As a workaround, you just have to check the old "Use Table" option, add a dummy parameter (mine's named "requiredPropertyToFixIssue4395"), and then check back the new "Use Map" option. The channel will then be valid again.

**Comment by arazak584** [comment]
> 
> 
> If someone has the same problem...
> 
> As a workaround, you just have to check the old "Use Table" option, add a dummy parameter (mine's named "requiredPropertyToFixIssue4395"), and then check back the new "Use Map" option. The channel will then be valid again.

Please I'm a novice in mirth connect, I will be glad if can guide me through how to resolve it, I am facing the same issue. My email is arazak584@yahoo.com

**Comment by is-simon** [comment]
@arazak584 Just check the "Use Table" option in order to enable the "add", "remove" options of the parameters table, then add a parameter (the name could be anything, it won't be used anyway). Then check back the new "Use Map" option. Now you will be able to save the channel, and since the parameter table is not empty (even if not used), the channel will be validated successfully.

**Comment by arazak584** [comment]
Thanks very much for the information but I’m using mirth connect 3.4.2 with the source as JAVASCRIPT READER and destination connector as HTTP SENDER

and I cannot locate “Use Table” Option

Thanks

**Comment by is-simon** [comment]
Oh I see...

Then in your URL, you could set something like this:

`http://example.com?${queryParams}
`

and in a transformer (JavaScript step), it would be something like this:
```
// Build the query parameters.
var queryParams = "param1=value1";
if (<some condition here>) {
	queryParams += "&param2=value2";
}
connectorMap.put("queryParams", queryParams);
```

But I'not sure how you faced the same issue as my original one, since this option wasn't available on Mirth 3.4.2

**Comment by cturczynskyj** [comment]
Fixed in 3.9.1.
